lib/apiservers/engine/backends/container.go

```diff
@@ -1543,6 +1546,72 @@ func ContainerSignal(containerID string, sig uint64) error {
 	return nil
 }
 
+func getClientIPAddrs() []netlink.Addr {
+
+	l, err := netlink.LinkByName(clientIfaceName)
+	if err != nil {
+		log.Errorf("Could not look up link from client interface name %s due to error %s",
+			clientIfaceName, err.Error())
+		return nil
+	}
+	ips, err := netlink.AddrList(l, netlink.FAMILY_V4)
+	if err != nil {
+		log.Errorf("Could not get IP addresses of link due to error %s", err.Error())
+		return nil
+	}
+
+	return ips
+}
+func getPortInformation(t *models.ContainerInfo, names []string) []types.Port {
+	if len(names) == 0 {
+		log.Infof("No container names provided; cannot get port information for unknown containers")
+		return nil
+	}
+
+	// create a port for each IP on the interface (usually only 1, if netlink.FAMILY_ALL then usually 2)
+	ips := getClientIPAddrs()
+	var ports []types.Port
+	for _, ip := range ips {
+		ports = append(ports, types.Port{IP: ip.IP.String()})
+	}
+	container := cache.ContainerCache().GetContainer(names[0])
+	if container == nil {
+		if match := containerFriendlyNamePattern.FindString(names[0]); match != "" {
```

Comment:
It actually always runs this piece. During the rebase yesterday the friendly name got a `/` prepended to it. Seems like a bug but I'm not sure. I did this rather than stripping the `/` to be robust in case the preceding `/` is removed.